bika/lims/browser/worksheet/ajax.py

```diff
@@ -67,11 +67,13 @@ def __call__(self):
             if analysis.portal_type in ('Analysis', 'DuplicateAnalysis'):
                 if review_state not in attachable_states:
                     continue
-                parent = analysis.getRequestID()
+                parent = analysis.getRequest().getId()
             elif analysis.portal_type == 'ReferenceAnalysis':
                 if review_state not in attachable_states:
                     continue
                 parent = analysis.aq_parent.Title()
+            else:
+                raise RuntimeError("Requires Duplicate/Analysis/Reference")
```

Comment:
Is ok, but now you are here... all this loop full of ifelses can be simplified a lot: 
```python
for analysis in analyses:
    review_state = workflow.getCurrentState(analysis)
    if review_state not in attachable_states:
        continue
    parent = analysis.getParentTitle()
    ...
```